app/src/main/java/io/zenandroid/onlinego/ui/screens/puzzle/TsumegoUI.kt

```diff
@@ -0,0 +1,249 @@
+package io.zenandroid.onlinego.ui.screens.puzzle
+
+import androidx.compose.animation.*
+import androidx.compose.foundation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.google.accompanist.pager.*
+import io.zenandroid.onlinego.R
+import io.zenandroid.onlinego.data.model.Cell
+import io.zenandroid.onlinego.data.model.StoneType
+import io.zenandroid.onlinego.ui.composables.Board
+import io.zenandroid.onlinego.ui.composables.RatingBar
+import io.zenandroid.onlinego.ui.screens.puzzle.TsumegoAction.*
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filter
+import org.commonmark.node.*
+import org.koin.core.context.GlobalContext
+import java.time.temporal.ChronoUnit.*
+
+private const val TAG = "TsumegoUI"
+
+@Composable
+fun TsumegoScreen(
+    state: TsumegoState,
+    hasPreviousPuzzle: Boolean,
+    hasNextPuzzle: Boolean,
+    onMove: (Cell) -> Unit,
+    onHint: () -> Unit,
+    onResetPuzzle: () -> Unit,
+    onRate: (Int) -> Unit,
+    onSolved: () -> Unit,
+    onPreviousPuzzle: () -> Unit,
+    onNextPuzzle: () -> Unit,
+    onBack: () -> Unit,
+) {
+    val solved = state.continueButtonVisible
+    Column (
+        modifier = Modifier.fillMaxHeight()
+    ) {
+        val titleState = remember {
```

Comment:
I haven't tried it, but I don't think remember and derivestateOf are needed here.